### PR TITLE
Fixes combined version

### PR DESCRIPTION
**Pull request by nikolase90**
Fixes https://github.com/NorskRegnesentral/shapr/issues/54.

### CODE REVIEW

File: R/explanation.R [hunk 1]
```diff
@@ -8,9 +8,9 @@
 #' @param explainer An \code{explainer} object to use for exaplaining the observations.
 #' See \code{\link{shapr}}.
 #'
-#' @param approach String or list. If a String, the approach to use when computing the Shapley
-#' values. Equal to 'empirical', 'gaussian' or 'copula'. If a list, the elements in the list
-#' refers to the rows in \code{x} that ought to be included in each of the approaches.
+#' @param approach Character. Note that \code{1 <= length(approach) <= n_features}, where
+#' \code{n_features} where equals the total number of features in the model. All elements should
```
Comment (martinju) [inline R/explanation.R]: typo: where \code{n_features} *where* equals

File: R/explanation.R [hunk 2]
```diff
@@ -8,9 +8,9 @@
 #' @param explainer An \code{explainer} object to use for exaplaining the observations.
 #' See \code{\link{shapr}}.
 #'
-#' @param approach String or list. If a String, the approach to use when computing the Shapley
-#' values. Equal to 'empirical', 'gaussian' or 'copula'. If a list, the elements in the list
-#' refers to the rows in \code{x} that ought to be included in each of the approaches.
+#' @param approach Character. Note that \code{1 <= length(approach) <= n_features}, where
```
Comment (martinju) [inline R/explanation.R]: Suggests:  @param approach Character (vector). Note that ....

File: R/explanation.R [hunk 3]
```diff
@@ -8,9 +8,9 @@
...
+#' \code{n_features} where equals the total number of features in the model. All elements should
+#' either be \code{gaussian}, \code{copula} or \code{empirical}. See details for more information.
```
Comment (martinju) [inline R/explanation.R]: shouldn't these be in quotes? "gaussian", "copula" etc.

File: R/explanation.R [hunk 4]
```diff
@@ -20,10 +20,10 @@
 #'
 #' @param seed Positive integer. If \code{NULL} a random seed will be used.
 #'
-#' @param ... Additional arguments passed to \code{explain.empirical}, \code{explain.gaussian} or
-#' \code{explain.copula}.
+#' @param ... Additional arguments passed to \code{\link{prepare_data}}
 #'
 #' @details
+#' TODO: Add information about approach.
```
Comment (martinju) [inline R/explanation.R]: Should be added now. Currently it is impossble to guess what happens when length(approach) != n_features

File: R/observations.R [hunk 5]
```diff
@@ -52,11 +55,23 @@ prepare_data <- function(x, ...) {
...
+    x$x_train,
+    x$x_test,
+    x$X$features[index_features]
+  )
 
   # Setup
+  if (!is.null(seed)) set.seed(seed)
   n_col <- nrow(x$x_test)
-  no_empirical <- nrow(x$S)
+  no_empirical <- nrow(x$S[index_features, ])
```
Comment (martinju) [inline R/observations.R]: I think you need drop=F here to handle cases where length(index_features)=1. Shouldn't a test have captured such a problem?

File: R/observations.R [hunk 6]
```diff
@@ -87,13 +102,13 @@ prepare_data.empirical <- function(x, seed = 1, n_samples = 1e3, ...) {
...
     if (kernel_metric == "independence") {
-      D <- D[sample(nrow(D)), ]
+      D <- D[sample.int(nrow(D)), ]
       h_optim_vec <- mean(D) * 1000
     }
 
     val <- t(t(-0.5 * D) / h_optim_vec^2)
     W_kernel <- exp(val)
-    S <- x$S
+    S <- x$S[index_features, ]
```
Comment (martinju) [inline R/observations.R]: I think the drop=F applies here as well.

File: R/explanation.R (hunk 1 above)
Comment (nikolase90) [reply R/explanation.R]: Fixed with 50295a1.

File: R/explanation.R (hunk 2 above)
Comment (nikolase90) [reply R/explanation.R]: Fixed with dee3db7.

File: R/explanation.R (hunk 3 above)
Comment (nikolase90) [reply R/explanation.R]: Fixed with 2fa0499.

File: R/observations.R (hunk 5 above)
Comment (nikolase90) [reply R/observations.R]: `index_features` should never be an element less than 2. I'll need to make an small change to make sure that this is the case.

Comment (nikolase90) [reply R/observations.R]: This should fix it. c7fe376

**Pull request merged**